filament/backend/include/private/backend/MetalPlatform.h

```diff
@@ -17,28 +17,36 @@
 #ifndef TNT_FILAMENT_DRIVER_PLATFORM_METAL_H
 #define TNT_FILAMENT_DRIVER_PLATFORM_METAL_H
 
-#include <stdint.h>
-
 #include <backend/DriverEnums.h>
 #include <backend/Platform.h>
 
+#import <Metal/Metal.h>
+
 namespace filament {
 namespace backend {
 
 class MetalPlatform : public DefaultPlatform {
 public:
     ~MetalPlatform() override;
-};
 
-} // namespace backend
-
-class PlatformMetal final : public backend::MetalPlatform {
-public:
-    backend::Driver* createDriver(void* sharedContext) noexcept override;
+    Driver* createDriver(void* sharedContext) noexcept override;
     int getOSVersion() const noexcept override { return 0; }
-    ~PlatformMetal() override;
+
+    virtual id<MTLDevice> createDevice() noexcept;
+    virtual id<MTLCommandQueue> createCommandQueue(id<MTLDevice> device) noexcept;
```

Comment:
Maybe some documentation comments?

Reply:
One interesting thing about this is API is that these calls do not occur in the driver thread.

Presumably this isn't a problem in Metal because there's no notion of "current context" like here is in GL.

(In Vulkan there's no notion of "current context" either, but users are still expected to to avoid calling certain methods from more than one thread.)

Just thought I'd mention it in case there's a gotcha we haven't thought of yet.

Reply:
I think `createDevice` is used by our driver and it's not really intended for outside parties, right?  Maybe we need a small note to warn users about that.

Reply:
No, these calls happen on the driver thread (they're called from the `MetalDriver` constructor. The idea is that outside parties could override `createDevice` and/or `createCommandQueue` if they wanted to.. the platform might want to select a specific device, for example.